netlogo-gui/src/main/app/WidgetActions.scala

```diff
@@ -0,0 +1,145 @@
+package org.nlogo.app
+
+import java.awt.Rectangle
+import java.awt.event.ActionEvent
+import javax.swing.{AbstractAction, JComponent}
+import javax.swing.undo.{AbstractUndoableEdit, UndoManager, UndoableEdit}
+
+import org.nlogo.core.Widget
+import org.nlogo.swing.ToolBarButton
+import org.nlogo.window.MouseMode
+
+object WidgetActions {
+
+  val undoManager = new org.nlogo.editor.UndoManager() {
+    // The default one doesn't work!
```

Comment:
Can you explain this comment? If you just mean that `canRedo` needs an override, let's just create an inner class for that. Otherwise, let's fix the problems with `org.nlogo.editor.UndoManager`.